crates/fuel-core/src/service/adapters/txpool.rs

```diff
@@ -102,9 +119,45 @@ impl fuel_core_txpool::ports::PeerToPeer for P2PAdapter {
             Ok(())
         }
     }
+
+    async fn request_tx_ids(&self, peer_id: Vec<u8>) -> anyhow::Result<Vec<TxId>> {
+        if let Some(service) = &self.service {
+            match service.get_all_transactions_ids_from_peer(peer_id).await {
+                Ok(txs) => Ok(txs),
+                Err(e) => {
+                    tracing::error!("Error getting tx ids from peer: {:?}", e);
+                    Ok(vec![])
```

Comment:
Why we just log the error but do not propagate it upwards?

Reply:
I don't want the `txpool` service to stop because of the error if it's just that we can't communicate with this peer anymore. Maybe this function shouldn't return a result at all.

Reply:
Done

Reply:
well, this function should return an error if sending the request to the peer fails, so I don't think we should remove the result from this.

Reply:
Done